src/components/panel/panelSections/panelHeader.component.tsx

```diff
@@ -13,41 +13,46 @@ import {
 import { Edit, Expand, Trash, MoreVertical } from "src/components/icons";
 import { PanelLabel } from "src/components/panel/panelLabel";
 import { Pill } from "src/components/pill";
+import { LanguageSelect } from "src/components/select";
 import { Toast } from "src/components/toast/toast.component";
 import { useDeleteObject } from "src/hooks/useDeleteObject";
-import { SkylarkObjectType } from "src/interfaces/skylark";
+import { ParsedSkylarkObject, SkylarkObjectType } from "src/interfaces/skylark";
+import { getObjectDisplayName } from "src/lib/utils";
 
 interface PanelHeaderProps {
-  title: string;
+  objectUid: string;
+  objectType: SkylarkObjectType;
+  object: ParsedSkylarkObject | null;
+  activeLanguage: string;
   currentTab: string;
   tabsWithEditMode: string[];
-  objectType: SkylarkObjectType;
-  objectUid: string;
-  pillColor?: string;
   graphQLQuery: DocumentNode | null;
   graphQLVariables?: object;
   inEditMode: boolean;
   isSaving?: boolean;
   toggleEditMode: () => void;
   closePanel?: () => void;
   save: () => void;
+  setActiveLanguage: Dispatch<SetStateAction<string>>;
 }
 
 export const PanelHeader = ({
-  title,
-  objectType,
   objectUid,
+  objectType,
+  object,
+  activeLanguage,
   currentTab,
   tabsWithEditMode,
-  pillColor,
   graphQLQuery,
   graphQLVariables,
   inEditMode,
   isSaving,
   toggleEditMode,
   closePanel,
   save,
+  setActiveLanguage,
 }: PanelHeaderProps) => {
+  const title = object ? getObjectDisplayName(object) : "";
```

Comment:
not for now, but probably we should have the option of the argument being `undefined`, and returning an empty string inside. otherwise, we are making these checks everywhere.

Reply:
yes agree!